sdk/dotnet/Pulumi/Deployment/Deployment_ReadOrRegisterResource.cs

```diff
@@ -46,16 +46,16 @@ public partial class Deployment
                     "pulumi:pulumi:getResource",
                     new GetResourceInvokeArgs {Urn = options.Urn},
                     new InvokeOptions()).ConfigureAwait(false);
-                
+
                 var urn = result.Fields["urn"].StringValue;
                 var id = result.Fields["id"].StringValue;
                 var state = result.Fields["state"].StructValue;
                 return (urn, id, state, ImmutableDictionary<string, ImmutableHashSet<Resource>>.Empty);
             }
-            
+
             if (options.Id != null)
             {
-                var id = await options.Id.ToOutput().GetValueAsync().ConfigureAwait(false);
+                var id = await options.Id.ToOutput().GetValueAsync(whenUnknown: "").ConfigureAwait(false);
```

Comment:
No change in behavior, similar to `default!` because of `string.IsNullOrEmpty` below.